WebPageProxy::MouseDidMoveOverElement Modifiers Serialization

abigailfox opened the pull request:
#### 5fe58ec5ffb752c5dfdcf71bb148e3107be98fc5
<pre>
WebPageProxy::MouseDidMoveOverElement Modifiers Serialization
<a href="https://bugs.webkit.org/show_bug.cgi?id=244557">https://bugs.webkit.org/show_bug.cgi?id=244557</a>
rdar://63357649

Reviewed by David Kilzer.

* Source/WebKit/Scripts/webkit/messages.py:
(headers_for_type):
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::mouseDidMoveOverElement):
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/UIProcess/WebPageProxy.messages.in:
* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.cpp:
(WebKit::WebChromeClient::mouseDidMoveOverElement):

Canonical link: <a href="https://commits.webkit.org/253988@main">https://commits.webkit.org/253988@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f3ef1b8de83cfce919d32019f5f3a5560db0d1f3

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/87661 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/69/builds/31747 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/18411 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/96862 "Built successfully") | 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/64/builds/30105 "Built successfully") | [  ~~🛠 mac-debug~~](https://ews-build.webkit.org/#/builders/71/builds/26226 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/79755 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/91613 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/93279 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/68/builds/24322 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/74416 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/24144 "Passed tests") | 
| [❌ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/5/builds/91265 "Passed webkitpy python3 tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/79285 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/79474 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/67159 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/27801 "Built successfully") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/13299 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/27766 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/14315 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/2805 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/29460 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/37208 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/29363 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/33592 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

ddkilzer commented on Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.cpp:
```diff
@@ -713,13 +713,15 @@ void WebChromeClient::unavailablePluginButtonClicked(Element& element, RenderEmb
 void WebChromeClient::mouseDidMoveOverElement(const HitTestResult& hitTestResult, unsigned modifierFlags, const String& toolTip, TextDirection)
 {
     RefPtr<API::Object> userData;
+    
```
Super-nit: Don't need a blank line here.  it's not against style, but usually variables are grouped together instead of spaced.

ddkilzer replied:
This doesn't need to be changed.  Added `merge-queue` keyword.

ddkilzer reviewed (approved):
r=me once tests pass.  Removing the blank line is optional.

abigailfox committed:
WebPageProxy::MouseDidMoveOverElement Modifiers Serialization
https://bugs.webkit.org/show_bug.cgi?id=244557
rdar://63357649

Reviewed by David Kilzer.

* Source/WebKit/Scripts/webkit/messages.py:
(headers_for_type):
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::mouseDidMoveOverElement):
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/UIProcess/WebPageProxy.messages.in:
* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.cpp:
(WebKit::WebChromeClient::mouseDidMoveOverElement):

Canonical link: https://commits.webkit.org/253988@main

ddkilzer commented on Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.cpp:
```diff
@@ -713,13 +713,15 @@ void WebChromeClient::unavailablePluginButtonClicked(Element& element, RenderEmb
 {
     RefPtr<API::Object> userData;
+    
+    auto modifiers = OptionSet<WebEvent::Modifier>::fromRaw(modifierFlags);
```
Just to follow-up on this patch—everything was correct except for this one change.

The `modifierFlags` parameter contains an `NSEventModifierFlags` value on macOS, and  `WebEventFlags` value on iOS (and any platform with UIKit mouse support), so this requires a separate conversion method instead of using `OptionSet<WebEvent::Modifier>::fromRaw()`.
